corehq/apps/locations/models.py

```diff
@@ -274,6 +274,15 @@ def couch_locations(self, wrapped=True):
             return itertools.imap(Location.wrap, locations)
         return locations
 
+    def assigned_to_user(self, domain, user):
+        if user.has_permission(domain, 'access_all_locations'):
+            return self.all()
+
+        assigned_location_ids = user.get_assigned_location_ids(domain)
+        if not assigned_location_ids:
+            return self.none()  # No locations are assigned to this user
+        return self.all() & SQLLocation.active_objects.get_locations_and_children(assigned_location_ids)
+
```

Comment:
I think this should go in `accessible_to_user` instead, since every usage of that should also account for multiple locations.  That method predates multiple locations per user, but part of the reason I abstracted that out was so we could change the meaning of "accessible to user" in the future and have everything using it automatically updated.